### DISCUSSION TITLE
Multiple categories with various number of keypoints

### DISCUSSION THREAD

**Discussion by i008** [open-mmlab/mmpose#766] [Q&A]
Hello, i have a case where my dataset consists of multiple classes with different keypoints (types and numer) Think of it as having a model that will do animal pose estimation for all the animal types.

How do i make it work in this framework? In my previous attempt (not using mmpose) i combined the keypoint labels so for instance
class-A: 5 keypoints
class B: 10 keypoints

And for each instance i would predict 15 keypoint maps - and just for the instances where keypoints are not relevant i would set visibility to false etc. The model would learn this association.

What would be the correct approach using mmpose and how to prepare the training_datasets and configs ?

**Comment by jin-s13** [comment]
One way might be 
1) For both class-A and class-B, transform the original labels to 15 keypoints. And for the missing keypoints, just leave them as [0, 0, 0]. 
2) Merge the two dataset label json files into one.
And now, you have obtained a new dataset with 15 keypoints.

**Reply by i008** [reply]
Thanks for the prompt answer and confirmation:) i will try this approach.
